docs/guides/extensions/operations-npm-package.md

```diff
@@ -0,0 +1,171 @@
+`directus-operation-lodash`). For this guide, select JavaScript.
+
+Now the operation has been created, go into the current directory, install the chosen npm package, and build the
+extension. This must be performed whenever your changes are ready to be deployed.
+
+```
+cd directus-operation-lodash
+npm install lodash
+npm run build
+```
+
+## Build the Operation UI
+
+Operations have 2 parts - the `api.js` file that performs logic, and the `app.js` file that describes the front-end UI
+for the operation.
+
+Open `app.js` and change the `id`, `name`, `icon`, and `description`.
+
+```js
+id: 'operation-lodash-camelcase',
+name: 'Lodash Camel Case',
+icon: 'electric_bolt',
+description: 'Use Lodash Camel Case Function.',
+```
+
+Make sure the `id` is unique between all extensions including ones created by 3rd parties - a good practice is to
+include a professional prefix. You can choose an icon from the library [here](https://fonts.google.com/icons).
+
+With the information above, the operation will appear in the list like this:
+
+<img src="https://marketing.directus.app/assets/8e5245cb-2454-4c46-8a59-04a325385d61.png" alt="Add comment - add a comment to a record. A chat icon is displayed in the box." style="padding:3px 2px 2px;"/>
+
+### Accepting Inputs
+
+The default `options` object in `app.js` has a single text interface called `text`:
+
+```js
+options: [
+	{
+		field: 'text',
+		name: 'Text',
+		type: 'string',
+		meta: {
+			width: 'full',
+			interface: 'input',
+		},
+	},
+],
+```
+
+If your npm package function requires multiple inputs, you can add them here. The `overview` array defines what is shown
+on the card when the operation is not selected.
+
+## Build the API Function
+
+Open the `api.js` file and update the `id` to match the one used in the `app.js` file. Import from your npm package,
+execute your logic, and finish by return any data from the operation into the data chain.
+
+```js
+import { defineOperationApi } from '@directus/extensions-sdk';
+import { camelCase } from 'lodash'; // [!code ++]
+
+export default defineOperationApi({
+	id: 'operation-lodash-camelcase',
+	handler: ({ text }) => {
+		console.log(text) // [!code --]
+		return { // [!code ++]
+			text: camelCase(text) // [!code ++]
+		} // [!code ++]
+	},
+});
+```
+
+Both files are now complete. Build the operation with the latest changes.
+
+```
+npm run build
+```
+
+## Add Operation to Directus
+
+In order to use this operation in Directus, you must copy the compiled index file into the project's extension folder.
+
+1. In the Directus extensions directory, open the operations directory and make a new directory called
+   `directus-operation-lodash`.
+2. From the operation's directory, open the **dist** folder and copy the `app.js` and `api.js` files into the directory.
+3. Install `lodash` helper library into your Directus root directory with `npm install lodash`.
```

Comment:
The dependencies should have gotten bundled with the build-step, making this unnecessary.